python/tmc5130.py

```diff
@@ -0,0 +1,1427 @@
+            spi.write_readinto(write_data, read_data)
+
+        # SPI Data; 8bit stats | MSB - LSB 32 bit data
+        status = read_data[0]
+        # Get bytes 1 to 4 and reverse them so we can use the to create python classes
+        data = read_data[1:][::-1]
+        return status, data
+
+    def init(self):
+        """
+        Configure the startup settings for TMC5130
+        """
+        # General Config
+        general_config = GeneralConfig()
+        chop_config = ChopConf()
+        pwm_mode_chop = PWMConf()
+        tpwm_threshold = TPWMTHRS()
+        cool_threshold = TCOOLTHRS()
+        power_down = TPowerDown()  # Power down in standstill
+        power_down.tpowerdown = 0x10
+        ramp_mode = RampMode()
+        cool_conf = CoolConf()
+
+        # Set the general config
+        general_config.shaft = 0
+        general_config.diag0_stall = 0
+        general_config.diag0_int_pushpull = 0
+        general_config.en_pwm_mode = 1  # Does not work with slow speeds
+
+        # Chopper Config
+        chop_config.toff = 0x03  # 0010
+        chop_config.hstrt = 0x04
+        chop_config.hend = 0x1
+        chop_config.tbl = 0x2  # Blank time
+        chop_config.chm = 0x01  # chopper mode
+        chop_config.vhighfs = 0x01  # Enable Fullstep at high velocity
+        chop_config.vhighchm = 0x01  # High velocity chopper mode
+
+        # Cool Step Config - Stallguard
+        cool_conf.sgt = 0x00
+
+        # PWM Conf
+        pwm_mode_chop.pwm_autoscale = 1
+        pwm_mode_chop.freewheel = 0x1  # 0 default, 1 free wheeling
+        pwm_mode_chop.pwm_ampl = 0x255
+        pwm_mode_chop.pwm_grad = 0x4
+
+        tpwm_threshold.tpwmthrs = 500
+
+        # Move type : ramp mode
+        ramp_mode.ramp_mode = self.config.get("ramp_mode", 0)
+
+        # Write to TMC Driver
+        self._write(GeneralConfig.address, general_config.get_register())
+        self._write(ChopConf.address, chop_config.get_register())
+        self._write(CoolConf.address, cool_conf.get_register())
+        self._write(PWMConf.address, pwm_mode_chop.get_register())
+        self._write(TPWMTHRS.address, tpwm_threshold.get_register())
+        self._write(TCOOLTHRS.address, cool_threshold.get_register())
+        self._write(TPowerDown.address, power_down.get_register())
+        self._write(RampMode.address, ramp_mode.get_register())
+        self.set_current(**self.config.get("current_mA", {}))
+        self.set_target_position(0)
+        self.get_current_position(0)
+        self.set_motion_rpm(velocity_rpm=0, acceleration_rpm=0, deceleration_rpm=0)
+
+    def set_current(self, run_mA: int, hold_mA: int):
+        """
+        Set the current for the TMC Driver
+        """
+        Rsense = self.config.rsense
+        # Convert from mA to ihold and irun
+        irun = 32.0 * 1.41421 * run_mA / 1000.0 * (Rsense + 0.02) / 0.325 - 1
+        # Limit ihold to 31 and 0
+        irun = min(31, max(0, irun))
```

Comment:
Should return a warning if `irun` is below 15. 16 to 31 is the optimal range.

Reply:
Is that in the docs?

Reply:
Done, added docs to explain how it works, in the future, we'll expose second set of acc decel and velocity (a1, v1, d1)